### PR TITLE
KW/CX 6pm-1 update office hour

### PR DESCRIPTION
**Pull request by kenny-wang6**
In this PR, the team added the feature for editing existing office hours, as well as front-end and back-end testing

### CODE REVIEW

File: javascript/src/main/components/OfficeHours/OfficeHourForm.js [hunk 1]
```diff
@@ -20,13 +20,18 @@ const OfficeHourForm = ({ createOfficeHour, /*updateOfficeHour, /*existingOffice
...
 
-    const [officeHour, setOfficeHour] = useState(emptyOfficeHour);
+    const [officeHour, setOfficeHour] = useState(existingOfficeHour||emptyOfficeHour);
 
     const handleOnSubmit = (e) => {
         e.preventDefault();
         var isValid = checkInputs(); 
         if (isValid) {
-            createOfficeHour(officeHour);
+            if (createOfficeHour) {
```
Comment (kaihilbourne) [inline javascript/src/main/components/OfficeHours/OfficeHourForm.js:29]: Is this how edit has been implemented for the other pages?

**Review by kaihilbourne — APPROVED** [review]
This looks good to me, and also like it was a lot of work. Probably worth a lot of points. I commented on the OfficeHourForm.js where I thought the code seemed like a quickfix type job, but I still think it is clear what it does and is good for its purpose.

**Comment by PureDreamerGk** [comment]
We are still trying to add a test for the office hour table, which verify that the edit button is rendered

File: javascript/src/main/components/OfficeHours/OfficeHourForm.js (hunk 1 above)
Comment (PureDreamerGk) [reply javascript/src/main/components/OfficeHours/OfficeHourForm.js:29]: Edit has been implemented similarly compared to other pages, but that does mean that it has the same implementation.

**Comment by kenny-wang6** [comment]
> We are still trying to add a test for the office hour table, which verify that the edit button is rendered

We fixed the issue and now have coverage for all tests.

File: javascript/src/main/components/OfficeHours/OfficeHourForm.js (hunk 1 above)
Comment (pconrad) [reply javascript/src/main/components/OfficeHours/OfficeHourForm.js:29]: This looks ok to me.    The `if (createOfficeHour)` is used to see whether the component that invoked this passed a value for `createOfficeHour` or not.  If they didn't, then it's assumed that they passed a value for `editOfficeHour`.   That's how the component knows whether it's being used for creating or editing.

**Review by wongjulian4213 — DISMISSED** [review]
Looks good to me!

**Review by kaihilbourne — APPROVED** [review]
Looks good to me! I am satisfied

**Review by pconrad — CHANGES_REQUESTED** [review]
There appears to be a bug.

STR: 
1.  Create a new office hour on Wednesday
2.  Try to edit it

Desired: When the screen comes up to edit, it should START with Wednesday as the day of the week.

Observed: When the screen comes up, it switches the office hour to Monday whether you like it or not.

**Comment by pconrad** [comment]
> There appears to be a bug.
> 
> STR:
> 
> 1. Create a new office hour on Wednesday
> 2. Try to edit it
> 
> Desired: When the screen comes up to edit, it should START with Wednesday as the day of the week.
> 
> Observed: When the screen comes up, it switches the office hour to Monday whether you like it or not.

Carl and I just working on this, and I think we fixed it.   Stay tuned.

**Review by pconrad — APPROVED** [review]
LGTM

**Review by kaihilbourne — APPROVED** [review]
LGTM!

**Pull request merged**